packages/smooth_app/lib/pages/product/attribute_icons.dart

```diff
@@ -0,0 +1,622 @@
+      excludeSemantics: true,
+      child: DecoratedBox(
+        decoration: ShapeDecoration(
+          shape: const CircleBorder(),
+          color: backgroundColor,
+        ),
+        child: Padding(
+          padding: paddingFactor != null
+              ? EdgeInsetsDirectional.only(
+                  start: paddingFactor!.start * size,
+                  end: paddingFactor!.end * size,
+                  top: paddingFactor!.top * size,
+                  bottom: paddingFactor!.bottom * size,
+                )
+              : EdgeInsets.zero,
+          child: Transform.translate(
+            offset: offsetFactor != null
+                ? Offset(
+                    size * offsetFactor!.dx,
+                    size * offsetFactor!.dy,
+                  )
+                : Offset.zero,
+            child: AppIconTheme(
+              color: foregroundColor,
+              size: size * (iconSizeFactor ?? 1.0),
+              child: icon,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    if (angle != null) {
+      child = Transform.rotate(
+        angle: angle!,
+        child: child,
+      );
+    }
+
+    if (clip == true) {
+      child = ClipOval(
+        child: child,
+      );
+    }
+
+    return SizedBox.square(
+      dimension: size,
+      child: child,
+    );
+  }
+}
+
+class _AttributeAdditivesIcon extends AttributeIcon {
+  const _AttributeAdditivesIcon({
+    required super.backgroundColor,
+    required super.size,
+    super.foregroundColor,
+    super.semanticsLabel,
+  }) : super._(
+          icon: const FoodIcons.additives(),
+          iconSizeFactor: 0.6,
+          paddingFactor: const EdgeInsetsDirectional.only(bottom: 0.05),
+        );
+}
+
+class _AttributeCeleryIcon extends AttributeIcon {
+  const _AttributeCeleryIcon({
+    required super.backgroundColor,
+    required super.size,
+    super.foregroundColor,
+    super.semanticsLabel,
+  }) : super._(
+          icon: const FoodIcons.celery(),
+          iconSizeFactor: 0.95,
+          offsetFactor: const Offset(-0.1, 0.0),
+          clip: true,
+          paddingFactor: const EdgeInsetsDirectional.only(top: 0.15),
+        );
+}
+
+class _AttributeCrustaceansIcon extends AttributeIcon {
+  const _AttributeCrustaceansIcon({
+    required super.backgroundColor,
+    required super.size,
+    super.foregroundColor,
+    super.semanticsLabel,
+  }) : super._(
+          icon: const FoodIcons.crustaceans(),
+          iconSizeFactor: 0.95,
+          clip: true,
+          paddingFactor: const EdgeInsetsDirectional.only(top: 0.15),
+        );
+}
+
+class _AttributeEggsIcon extends AttributeIcon {
+  const _AttributeEggsIcon({
+    required super.backgroundColor,
+    required super.size,
+    super.foregroundColor,
+    super.semanticsLabel,
+  }) : super._(
+          icon: const FoodIcons.eggs(),
+          iconSizeFactor: 0.65,
+          paddingFactor: const EdgeInsetsDirectional.only(top: 0.01),
+        );
+}
```

Comment:
```suggestion
class _AttributeFoodIcon extends AttributeIcon {
  const _AttributeFoodIcon({
    required super.backgroundColor,
    required super.size,
    super.foregroundColor,
    super.semanticsLabel,
    final FoodIcons icon,
  }) : super._(
          icon: icon,
          iconSizeFactor: icon.iconSizeFactor,
          paddingFactor: icon.paddingFactor,
          clip: icon.clip,
          offsetFactor: icon.offsetFactor,
          angle: icon.angle,
        );
}
```
Just checking: would that work? Then you wouldn't need all the classes.

Reply:
Your code is correct.
If we want better perf with Flutter, it's always better to have dedicated widgets

Reply:
I don't believe those tiny perf improvements have a significant impact on the overall perf of Smoothie, and I would prefer a more maintainable code.
Your code does work, though.